haskell/ghc_bindist.bzl

```diff
@@ -96,6 +98,13 @@ def _ghc_bindist_impl(ctx):
         _execute_fail_loudly(ctx, ["./configure", "--prefix", bindist_dir.realpath])
         _execute_fail_loudly(ctx, ["make", "install"])
 
+    # We apply some patches, if needed. As the patches may touch the package DB
+    # we regenerate the cache.
+    patch(ctx)
```

Comment:
Doesn't matter on Windows, but in general, what's the reason for patching after `./configure`, `make` instead of before?